File: backend/internal/domain/maturity.go
package domain

import "time"

type MetricType string

const (
	MetricTypeTestCoverage      MetricType = "test_coverage"
	MetricTypeDeployVelocity    MetricType = "deploy_velocity"
	MetricTypeMTTR              MetricType = "mttr"
	MetricTypeChangeFailureRate MetricType = "change_failure_rate"
	MetricTypeBuildQueueTime    MetricType = "build_queue_time"
	MetricTypePerformanceRegress MetricType = "performance_regression"
)

type MaturityCategory string

const (
	MaturityCategoryObservability    MaturityCategory = "observability"
	MaturityCategoryAutomatedTests   MaturityCategory = "automated_tests"
	MaturityCategoryIncidentResponse MaturityCategory = "incident_response"
	MaturityCategoryFinOps           MaturityCategory = "finops"
	MaturityCategorySecurity         MaturityCategory = "security"
	MaturityCategoryDocumentation    MaturityCategory = "documentation"
)

type ServiceMetric struct {
	ID          string                 `json:"id"`
	ServiceName string                 `json:"serviceName"`
	PRNumber    string                 `json:"prNumber,omitempty"`
	Type        MetricType             `json:"type"`
	Value       float64                `json:"value"`
	Unit        string                 `json:"unit"`
	Timestamp   time.Time              `json:"timestamp"`
	Metadata    map[string]interface{} `json:"metadata,omitempty"`
}

type MaturityScore struct {
	Category     MaturityCategory      `json:"category"`
	Score        float64               `json:"score"`        // 0-10
	MaxScore     float64               `json:"maxScore"`    // Always 10
	Level        string                `json:"level"`       // "beginner", "intermediate", "advanced", "expert"
	Metrics      []ServiceMetric       `json:"metrics"`
	Recommendations []Recommendation   `json:"recommendations"`
	LastUpdated  time.Time             `json:"lastUpdated"`
}

type TeamMaturityScorecard struct {
	TeamID       string                 `json:"teamId"`
	TeamName     string                 `json:"teamName"`
	Scores       []MaturityScore        `json:"scores"`
	OverallScore float64                `json:"overallScore"` // Average of all categories
	Rank         int                    `json:"rank,omitempty"`
	LastUpdated  time.Time              `json:"lastUpdated"`
	Trend        string                 `json:"trend"` // "improving", "stable", "declining"
}

type ServiceMaturityScorecard struct {
	ServiceName  string                 `json:"serviceName"`
	Scores       []MaturityScore        `json:"scores"`
	OverallScore float64                `json:"overallScore"`
	LastUpdated  time.Time              `json:"lastUpdated"`
}

type MaturityRecommendation struct {
	Category    MaturityCategory `json:"category"`
	Priority    string           `json:"priority"` // "high", "medium", "low"
	Title       string           `json:"title"`
	Description string           `json:"description"`
	Action      string           `json:"action"`
	Impact      string           `json:"impact"`
	Effort      string           `json:"effort"` // "low", "medium", "high"
}

File: mocks/src/mocks/data/maturity.ts
export interface Recommendation {
  id: string
  type: string
  severity: string
  title: string
  description: string
  reason: string
  action: string
  impact: string
  confidence: number
}

export interface MaturityScore {
  category: string
  score: number
  maxScore: number
  level: string
  metrics: any[]
  recommendations: Recommendation[]
  lastUpdated: string
}

export interface TeamScorecard {
  teamId: string
  teamName: string
  scores: MaturityScore[]
  overallScore: number
  rank?: number
  lastUpdated: string
  trend: string
}

const mockScorecards: Record<string, TeamScorecard> = {
  'Platform': {
    teamId: 'platform',
    teamName: 'Platform Team',
    overallScore: 7.8,
    rank: 1,
    lastUpdated: new Date().toISOString(),
    trend: 'improving',
    scores: [
      {
        category: 'observability',
        score: 8.5,
        maxScore: 10,
        level: 'advanced',
        metrics: [],
        recommendations: [
          {
            id: 'rec-1',
            type: 'observability',
            severity: 'medium',
            title: 'Implementar distributed tracing',
            description: 'Adicionar tracing distribuído para melhor visibilidade de requisições entre serviços',
            reason: 'Falta de visibilidade em requisições cross-service',
            action: 'Integrar OpenTelemetry ou Jaeger',
            impact: 'Melhor debugging e performance analysis',
            confidence: 85
          }
        ],
        lastUpdated: new Date().toISOString()
      },
      {
        category: 'automated_tests',
        score: 8.0,
        maxScore: 10,
        level: 'advanced',
        metrics: [],
        recommendations: [],
        lastUpdated: new Date().toISOString()
      },
      {
        category: 'incident_response',
        score: 7.5,
        maxScore: 10,
        level: 'intermediate',
        metrics: [],
        recommendations: [
          {
            id: 'rec-2',
            type: 'incident_response',
            severity: 'high',
            title: 'Criar runbooks automatizados',
            description: 'Documentar e automatizar procedimentos de resposta a incidentes comuns',
            reason: 'Tempo de resposta a incidentes pode ser reduzido',
            action: 'Criar runbooks e integrar com sistema de alertas',
            impact: 'Redução de 40% no MTTR',
            confidence: 90
          }
        ],
        lastUpdated: new Date().toISOString()
      },
      {
        category: 'finops',
        score: 7.0,
        maxScore: 10,
        level: 'intermediate',
        metrics: [],
        recommendations: [],
        lastUpdated: new Date().toISOString()
      },
      {
        category: 'security',
        score: 8.2,
        maxScore: 10,
        level: 'advanced',
        metrics: [],
        recommendations: [],
        lastUpdated: new Date().toISOString()
      },
      {
        category: 'documentation',
        score: 6.5,
        maxScore: 10,
        level: 'intermediate',
        metrics: [],
        recommendations: [
          {
            id: 'rec-3',
            type: 'documentation',
            severity: 'low',
            title: 'Melhorar documentação de APIs',
            description: 'Adicionar exemplos e casos de uso na documentação das APIs',
            reason: 'Documentação atual é básica',
            action: 'Expandir documentação OpenAPI com exemplos',
            impact: 'Melhor onboarding e uso das APIs',
            confidence: 75
          }
        ],
        lastUpdated: new Date().toISOString()
      }
    ]
  },
  'Security': {
    teamId: 'security',
    teamName: 'Security Team',
    overallScore: 8.5,
    rank: 1,
    lastUpdated: new Date().toISOString(),
    trend: 'improving',
    scores: [
      {
        category: 'security',
        score: 9.5,
        maxScore: 10,
        level: 'expert',
        metrics: [],
        recommendations: [],
        lastUpdated: new Date().toISOString()
      },
      {
        category: 'observability',
        score: 8.0,
        maxScore: 10,
        level: 'advanced',
        metrics: [],
        recommendations: [],
        lastUpdated: new Date().toISOString()
      },
      {
        category: 'automated_tests',
        score: 8.8,
        maxScore: 10,
        level: 'advanced',
        metrics: [],
        recommendations: [],
        lastUpdated: new Date().toISOString()
      },
      {
        category: 'incident_response',
        score: 8.2,
        maxScore: 10,
        level: 'advanced',
        metrics: [],
        recommendations: [],
        lastUpdated: new Date().toISOString()
      },
      {
        category: 'finops',
        score: 7.5,
        maxScore: 10,
        level: 'intermediate',
        metrics: [],
        recommendations: [],
        lastUpdated: new Date().toISOString()
      },
      {
        category: 'documentation',
        score: 8.0,
        maxScore: 10,
        level: 'advanced',
        metrics: [],
        recommendations: [],
        lastUpdated: new Date().toISOString()
      }
    ]
  },
  'Payments': {
    teamId: 'payments',
    teamName: 'Payments Team',
    overallScore: 6.8,
    rank: 3,
    lastUpdated: new Date().toISOString(),
    trend: 'declining',
    scores: [
      {
        category: 'security',
        score: 7.0,
        maxScore: 10,
        level: 'intermediate',
        metrics: [],
        recommendations: [
          {
            id: 'rec-4',
            type: 'security',
            severity: 'critical',
            title: 'Corrigir vulnerabilidades críticas',
            description: 'Existem vulnerabilidades críticas que precisam ser corrigidas urgentemente',
            reason: 'Vulnerabilidades detectadas no código',
            action: 'Revisar e corrigir vulnerabilidades identificadas',
            impact: 'Redução de risco de segurança',
            confidence: 95
          }
        ],
        lastUpdated: new Date().toISOString()
      },
      {
        category: 'automated_tests',
        score: 6.5,
        maxScore: 10,
        level: 'intermediate',
        metrics: [],
        recommendations: [
          {
            id: 'rec-5',
            type: 'automated_tests',
            severity: 'high',
            title: 'Aumentar cobertura de testes',
            description: 'Cobertura atual está abaixo do ideal para um serviço crítico',
            reason: 'Cobertura de testes em 65%',
            action: 'Adicionar testes unitários e de integração',
            impact: 'Maior confiabilidade do serviço',
            confidence: 88
          }
        ],
        lastUpdated: new Date().toISOString()
      },
      {
        category: 'observability',
        score: 7.2,
        maxScore: 10,
        level: 'intermediate',
        metrics: [],
        recommendations: [],
        lastUpdated: new Date().toISOString()
      },
      {
        category: 'incident_response',
        score: 6.0,
        maxScore: 10,
        level: 'intermediate',
        metrics: [],
        recommendations: [],
        lastUpdated: new Date().toISOString()
      },
      {
        category: 'finops',
        score: 6.8,
        maxScore: 10,
        level: 'intermediate',
        metrics: [],
        recommendations: [],
        lastUpdated: new Date().toISOString()
      },
      {
        category: 'documentation',
        score: 6.5,
        maxScore: 10,
        level: 'intermediate',
        metrics: [],
        recommendations: [],
        lastUpdated: new Date().toISOString()
      }
    ]
  }
}

export const getMockMaturityScorecard = async (teamName: string): Promise<TeamScorecard> => {
  await new Promise(resolve => setTimeout(resolve, 300))
  const normalizedName = teamName.trim()
  
  if (mockScorecards[normalizedName]) {
    return mockScorecards[normalizedName]
  }
  
  if (normalizedName.toLowerCase().includes('platform')) {
    return mockScorecards['Platform']
  }
  
  if (normalizedName.toLowerCase().includes('security') || normalizedName.toLowerCase().includes('auth')) {
    return mockScorecards['Security']
  }
  
  if (normalizedName.toLowerCase().includes('payment')) {
    return mockScorecards['Payments']
  }
  
  return mockScorecards['Platform']
}


